src/iso19111/factory.cpp

```diff
@@ -7293,6 +7293,24 @@ AuthorityFactory::getAuthorityCodes(const ObjectType &type,
 
 // ---------------------------------------------------------------------------
 
+/** \brief Returns the set of celestial body codes of the given object type.
+ *
+ * @return the set of celestial body codes
+ * @throw FactoryException
+ */
+std::set<std::string> AuthorityFactory::getCelestialBodyCodes() const {
+    std::string sql = "SELECT code FROM celestial_body "
```

Comment:
d->authority() might be empty when we want to query for all authorities. See AuthorityFactory::getUnitList() as how to handle that situation.

Reply:
Do you think that it would be useful for an object with authority, code, and name properties be returned then?

Reply:
It could be done, but this is more work. It would make sense if we anticipated that our description of a celestial body goes more than just a simple name. Well, currently the database table also stores a rough semi_major_axis value so that we can use it to figure out to which body an ellipsoid belongs too,as this information isn't stored anywhere in ISO-191111, WKT, etc.).  Also I'm a bit hesitant to expose more the concept (I'm not an astro geophysisit) than what is strictly needed and while more people have thought about the concept and how it relates / is different from an ellipsoid description.

The use cases I've in mind are:
- be able to filter CRS by bodies (nominal case: I want only Earth related CRS) ==> already covered
- given a CRS object, be able to determine to which body it belongs too ==> already covered
- list the allowed bodies: purpose of that PR

Reply:
The reason I ask is if we allow filtering on NULL, then it would be nice to know the authority it belongs to. Maybe authority, name, and semi_major_axis (exclude code)?

Reply:
> The reason I ask is if we allow filtering on NULL, then it would be nice to know the authority it belongs to. Maybe authority, name, and semi_major_axis (exclude code)?

I'm still hesitant about exposing too much, and would like to see some real use case we need to address. It is unlikely that any authority will actually deliver any code for such object, since it is absent from known models. This is more a PROJ artifact resulting from the need for a row in database to be identified by a code, and to have the ellipsoid table relates to it through a foreign key. I wouldn't really very enthusiastic to see the semi_major_axis to be exposed. The one used for Earth is some rounded value of an """"average""""" value, so that all known Earth ellipsoid match it (+/- 0.5 % I believe). Not something strongly scientificly founded.
As a compromise, if we believe that we might need more than just the name in the future, the function might return a list of
```
struct PROJ_CELESTIAL_BODY
{
     char* pszName;
     /* other fields might be added in the future */
}
```
(the working would then be like proj_get_units_from_database() )

Reply:
> I'm still hesitant about exposing too much, and would like to see some real use case we need to address.

I understand the part about exposing too much. I think authortiy and name should be sufficient in the return. It will provide clarity to the user when filtering that they got the right response and will also enable testing the filtering in the responses.

My main thought for this was to use this information to filter for CRS with the query crs info from database. This is something I think is very important if you only want CRS on `Earth` for example.

Reply:
> I think authortiy and name should be sufficient in the return

ok.